Epam/Java Basics_UpSkill Lab #1/Strings_and_basics_of_text_processing/Working_with_a_string_as_a_String_or_StringBuilder/src/com/androsor/string_object/HomeTask7.java

```diff
@@ -0,0 +1,58 @@
+package com.androsor.string_object;
+
+/**
+ * A string is entered. It is required to remove duplicate characters and all spaces from it. For example,
+ * if "abc cde def" was entered, then "abcdef" should be output.
+ */
+
+public class HomeTask7 {
+
+    public static void main (String [] arg){
+
+        String str;
+
+        str = HomeTask1.enterFromConsole("Введите строку:");
+
+        String newString = deleteRepeat(str);
+
+        System.out.printf("Строка без пробелов и повторяющихся символов\n%s", newString);
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [FS: Format string should use %n rather than n (VA_FORMAT_STRING_USES_NEWLINE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#fs-format-string-should-use-n-rather-than-n-va-format-string-uses-newline).
This format string includes a newline character (\n). In format strings, it is generally  preferable to use %n, which will produce the platform-specific line separator.